### PR TITLE
python3Packages: fix a bunch of broken packages

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
fixes builds

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

```
https://github.com/NixOS/nixpkgs/pull/89376
4 packages failed to build:
python27Packages.zope_i18n python37Packages.rl-coach python37Packages.zope_i18n python38Packages.zope_i18n

102 packages built:
buildbot buildbot-full buildbot-ui certbot cura devpi-client devpi-server gitAndTools.bump2version mailman octoprint python27Packages.celery python27Packages.clickclick python27Packages.connexion python27Packages.flower python27Packages.zope-hookable python27Packages.zope_component python27Packages.zope_configuration python27Packages.zope_i18nmessageid python27Packages.zope_lifecycleevent python27Packages.zope_size python37Packages.celery python37Packages.clickclick python37Packages.connexion python37Packages.cornice python37Packages.django-raster python37Packages.django-simple-captcha python37Packages.djmail python37Packages.flower python37Packages.mailman python37Packages.mailman-hyperkitty python37Packages.minio python37Packages.praw python37Packages.prawcore python37Packages.pyjade python37Packages.pyramid python37Packages.pyramid_beaker python37Packages.pyramid_chameleon python37Packages.pyramid_exclog python37Packages.pyramid_hawkauth python37Packages.pyramid_jinja2 python37Packages.pyramid_mako python37Packages.pyramid_multiauth python37Packages.scrapy python37Packages.sentry-sdk python37Packages.sopel python37Packages.stups-cli-support python37Packages.stups-fullstop python37Packages.stups-pierone python37Packages.stups-zign python37Packages.testfixtures python37Packages.zope-hookable python37Packages.zope_component python37Packages.zope_configuration python37Packages.zope_i18nmessageid python37Packages.zope_lifecycleevent python37Packages.zope_size python38Packages.buildbot python38Packages.buildbot-full python38Packages.buildbot-ui python38Packages.celery python38Packages.clickclick python38Packages.cornice python38Packages.django-raster python38Packages.django-simple-captcha python38Packages.djmail python38Packages.flower python38Packages.mailman python38Packages.mailman-hyperkitty python38Packages.minio python38Packages.praw python38Packages.prawcore python38Packages.pyjade python38Packages.pyramid python38Packages.pyramid_beaker python38Packages.pyramid_chameleon python38Packages.pyramid_exclog python38Packages.pyramid_hawkauth python38Packages.pyramid_jinja2 python38Packages.pyramid_mako python38Packages.pyramid_multiauth python38Packages.scrapy python38Packages.sopel python38Packages.stups-cli-support python38Packages.stups-fullstop python38Packages.stups-pierone python38Packages.stups-zign python38Packages.testfixtures python38Packages.zope-hookable python38Packages.zope_component python38Packages.zope_configuration python38Packages.zope_i18nmessageid python38Packages.zope_lifecycleevent python38Packages.zope_size sourcehut.buildsrht sourcehut.dispatchsrht sourcehut.gitsrht sourcehut.hgsrht sourcehut.listssrht sourcehut.mansrht sourcehut.metasrht sourcehut.pastesrht sourcehut.todosrht
```

### CODE REVIEW

**Comment by jonringer** [comment]
fixes ~102 builds

File: pkgs/development/python-modules/minio/default.nix [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
+    configparser
+    future
+    python-dateutil
+    pytz
+    urllib3
+  ];
 
-  checkInputs = [ faker mock nose ];
-  propagatedBuildInputs = [ urllib3 python-dateutil pytz future ];
+  checkInputs = [ faker mock nose pytest ];
```
Comment (Mic92) [inline pkgs/development/python-modules/minio/default.nix:31]:
```suggestion
  checkInputs = [ faker mock nose pytestCheckHook ];
```

Comment (jonringer) [reply pkgs/development/python-modules/minio/default.nix:31]: was holding out to see if i had to do some gnarly --ignore or -k arguments, but it works fine with just the vanilla invocation, i can change it

Comment (Mic92) [reply pkgs/development/python-modules/minio/default.nix:31]: at least -k can be replaced with `disabledTest`.

File: pkgs/development/tools/devpi-server/default.nix [hunk 2]
```diff
@@ -32,12 +32,17 @@ python3Packages.buildPythonApplication rec {
     webtest
   ] ++ stdenv.lib.optionals isPy27 [ mock ];
 
-  # test_genconfig.py needs devpi-server on PATH
   # root_passwd_hash tries to write to store
+  # TestMirrorIndexThings tries to write to /var through ngnix
+  # nginx tests try to write to /var
```
Comment (makefu) [inline pkgs/development/tools/devpi-server/default.nix:37]: that stuff is super creepy!

Comment (jonringer) [reply pkgs/development/tools/devpi-server/default.nix:37]: nginx whats to open a `/var/log/nginx/error.log`, which kind of makes sense

Comment (makefu) [reply pkgs/development/tools/devpi-server/default.nix:37]: "kind of", haha. i am still trying to build your PR with `nix-review`, it takes some time

File: pkgs/development/python-modules/celery/default.nix [hunk 3]
```diff
@@ -20,9 +22,13 @@ buildPythonPackage rec {
...
+  # test_multi tries to create directories under /var
   checkPhase = ''
-    pytest -k 'not restore_current_app_fallback and not msgpack and not on_apply' \
+    pytest -k 'not restore_current_app_fallback and not msgpack and not on_apply and not pytest' \
+      --ignore=t/unit/contrib/test_pytest.py \
       --ignore=t/unit/concurrency/test_eventlet.py \
+      --ignore=t/unit/bin/test_multi.py \
+      --ignore=t/unit/apps/test_multi.py \
       --ignore=t/unit/backends/test_mongodb.py
   '';
```
Comment (arnoldfarkas) [inline pkgs/development/python-modules/celery/default.nix:34]: cannot comment below, woudl it worth adding a maintainer for this package?

File: pkgs/development/python-modules/flower/default.nix [hunk 4]
```diff
@@ -1,22 +1,42 @@
...
-  
+  # flower and humanize aren't listed in setup.py but imported
+  propagatedBuildInputs = [
+    Babel
+    celery
+    future
+    importlib-metadata
+    pytz
+    tornado
+    humanize
```
Comment (arnoldfarkas) [inline pkgs/development/python-modules/flower/default.nix:31]: [fancy]
should it go above `pytz` ?

**Review by arnoldfarkas — APPROVED** [review]

**Review by makefu — APPROVED** [review]
devpi-client and devpi-server work as expected after running `nix-review pr 89376 --package-regex devpi\*`

thanks for your effort!

File: pkgs/development/python-modules/flower/default.nix (hunk 4 above)
Comment (jonringer) [reply pkgs/development/python-modules/flower/default.nix:31]: I forgot to sort, but it wasn't sorted before. I'm more concerned about the build failures than formatting :). doing small fixes on 10 different packages (and maintaining git practices) is a little painful

**Comment by jonringer** [comment]
at the very least, this fixes some builds. If there's improvements to be made, they can be done in another PR

**Pull request merged**